### DISCUSSION TITLE
Import from GNUCash

### DISCUSSION THREAD

**Discussion by adyanth** [firefly-iii/firefly-iii#6568] [Firefly III] [answered]
Is there any way to import transactions (multicurrency) from GNUcash's gnca format export? It looks like an XML file.

Edit: I was able to use the below script to make the XML style backup to a JSON structure (by no means complete or tested).
How do I go about moving this to firefly? Can I use the FIDI or just use the regular Firefly APIs to do this?

~https://gist.github.com/adyanth/2cedfbc5acd5a198c852e382f94637a6~
https://gist.github.com/adyanth/20c004869baf33458e416d4396ca40a8

**Comment by JC5** [comment]
FIDI only supports CSV files at the moment.

**Comment by adyanth** [comment, marked as answer]
Wrote a barebones gnca to firefly importer here: https://gist.github.com/adyanth/20c004869baf33458e416d4396ca40a8 (First part to convert gnca to json and the second one uses firefly APIs to enable currencies, import accounts and transactions)

Converted my 3 month full export from GNUCash without issues :)

**Reply by JC5** [reply]
Very cool! Is it OK if I list it in the documentation?

**Reply by adyanth** [reply]
Sure, go ahead! Although do add a disclaimer saying it is untested. If people are interested, I can convert it to a repository for collaboration.

**Reply by JC5** [reply]
Will do :+1:
